### ISSUE TITLE
[help wanted with double-check] matrix in .X, expect log1p normalized expression to 10000 counts per cell

### ISSUE THREAD

**Issue by KunHHE** [Teichlab/celltypist#144]
Hi @ChuanXu1 Very cool tool. Trying this tool for annotation, But I have a confused question about "matrix in .X, expect log1p normalized expression to 10000 counts per cell", want to double-check. So we I import my spatial Anndata, I only need to do sc.pp.normalize_total(adata, target_sum=1e4) -> sc.pp.log1p(adata), no more PCA/UMAP processing, right? And go to Celltypist directly?
Then If I used an annotated reference.h5ad, this is a annotated data, so used it as it is? Or I need to do anything else processing? Linking to similar issue#83, Thank you!

**Comment by ChuanXu1** [comment]
@KunHHE, yes, if you start by raw count data with all genes, you can normalize it with the two commands you have shown and directly go for CellTypist. CellTypist prediction does not rely on your own annotation, but you can visualize their correspondence using [celltypist.dotplot](https://celltypist.readthedocs.io/en/latest/celltypist.dotplot.html)

**Comment by KunHHE** [comment]
Hi @ChuanXu1 Thanks so much for your quick response! Can I ask you, once finishing the annotation by Celltypist,
how to process the data, Because the rwad data was process to be sc.pp.normalize_total(adata, target_sum=1e4) and sc.pp.log1p(adata).

We usually do it as below, right?

sc.pp.normalize_total(comb_adata, inplace=True)
sc.pp.log1p(comb_adata)
sc.pp.pca(comb_adata)
sc.pp.neighbors(comb_adata)
sc.tl.umap(comb_adata)
sc.tl.leiden(comb_adata)

Then UMAP....

**Comment by ChuanXu1** [comment]
@KunHHE, for your adata, CellTypist only adds additional columns in the `.obs`. So you can process your data without considering how CellTypist influences your data. For your code, you can perform HVGs detection and scaling between sc.pp.log1p and  sc.pp.pca
